airbyte-integrations/connectors/destination-snowflake/src/test-integration/kotlin/io/airbyte/integrations/destination/snowflake/component/SnowflakeTestTableOperationsClient.kt

```diff
@@ -71,15 +73,51 @@ class SnowflakeTestTableOperationsClient(
+                                                    // which has a toZonedDateTime function
+                                                    is SnowflakeTimestampWithTimezone ->
+                                                        it.toZonedDateTime().toOffsetDateTime()
+                                                    // Some timestamps are returned as
+                                                    // java.sql.Timestamp,
+                                                    // so we just assume UTC.
+                                                    is Timestamp ->
+                                                        it.toLocalDateTime()
+                                                            .atOffset(ZoneOffset.UTC)
+                                                }
+                                            row[columnName] =
+                                                DateTimeFormatter.ISO_DATE_TIME.format(odt)
+                                        }
+                                    }
+                                    "TIMESTAMPNTZ" -> {
+                                        resultSet.getTimestamp(i)?.let {
+                                            row[columnName] = it.toLocalDateTime()
+                                        }
+                                    }
+                                    "TIME" -> {
+                                        // resultSet.getObject (and .getTime) returns a
+                                        // java.sql.Time object, which only stores milliseconds
+                                        // precision.
+                                        // Snowflake supports up to nanoseconds precision, so we
+                                        // need retrieve the object as a timestamp, and convert that
+                                        // back to LocalTime.
+                                        resultSet.getTimestamp(i)?.let {
+                                            row[columnName] =
+                                                it.toLocalDateTime().toLocalTime().toString()
+                                        }
+                                    }
+                                    "DATE" -> {
+                                        // resultSet.getObject returns a java.sql.Date object,
+                                        // but its underlying `cdate` value is offset by someone's
+                                        // local timezone (unclear whether that's our snowflake user
+                                        // timezone, or the TZ of the computer running the test),
+                                        // and that breaks equality comparison.
+                                        // So just convert to String.
+                                        resultSet.getDate(i)?.let {
+                                            row[columnName] = it.toString()
```

Comment:
Nit: In general I prefer just putting the literal values in the expected values instead of adding more munging code, but ymmv

Reply:
chose not to do that here, mostly b/c I wasn't sure how the timezone thing actually worked (i.e. if I hardcode utc-8, does that break after the daylight savings switch). Generally agree though.